tasks/main.yml

```diff
@@ -40,16 +40,60 @@
       list | length > 0 }}"
     __firewall_python_cmd: "{{ ansible_python_interpreter |
       d(discovered_interpreter_python) }}"
+    __firewall_report_changed: true
 
-- name: Get config files, checksums before and remove
-  script:
-    cmd: >
-      files/get_files_checksums.sh {{ __firewall_python_cmd | quote }}
-      {{ __firewall_firewalld_dir | quote }} true
-  register: __firewall_config_files_before
-  changed_when: false
+- name: Handle previous replaced
   when: __firewall_previous_replaced | bool
+  block:
+    - name: Get config files, checksums before and remove
+      script:
+        cmd: >
+          files/get_files_checksums.sh {{ __firewall_python_cmd | quote }}
+          {{ __firewall_firewalld_dir | quote }} true
+          {{ __firewall_package_with_conf }}
+      register: __firewall_config_files_before
+      changed_when: false
+      when: not ansible_check_mode
+
+    - name: Check config files to see if they would be changed
+      find:
+        paths:
+          - "{{ __firewall_firewalld_dir }}"
+        patterns:
+          - "*.xml"
+      register: __firewall_files_found
+      changed_when: __firewall_files_found.matched > 0
+      when: ansible_check_mode | bool
```

Comment:
NetworkManager integration makes some changes, so we'll need to find a workaround for that if there is anything executing firewall-cmd after this task

Reply:
I now compare the checksum of the file under /etc/firewalld with the file under /usr/lib/firewalld